### DISCUSSION TITLE
Maximum Number of Namespaces/Rooms

### DISCUSSION THREAD

**Discussion by vsaw** [socketio/socket.io#4754] [Q&A] [answered]
I'm considering using Socket.IO for an IoT application where I want devices to be able to send telemetry to a server and receive commands from a server.

Since telemetry and commands will be handled by different parts of the application, I'm considering using Namespaces for this. The different namespaces would be something like: `<device-uuid>/telemetry` and `<device-uuid>/commands` (e.g. `19fddd85-a88c-4016-85d7-bffca02577b2/telemetry`)

On the server I would create a matching wildcard to handle messages and connections from devices.

I guess my questions are the following

1. Is that a sensible use of Namespaces or would Rooms be better suited for this?
2. Are there any hard/soft limits on the number of Namespaces or Rooms Socket.IO can sensibly handle?

Thank you 🙏

**Comment by darrachequesne** [comment, marked as answer]
Hi!

> Is that a sensible use of Namespaces or would Rooms be better suited for this?

Without knowing all details of your application, it's a bit hard to tell, but maybe one namespace for telemetry and one for commands would suit your need:

```js
io.of("/telemetry").on("connection", (socket) => {
  socket.join(<device-uuid>);

  io.to(<device-uuid>).emit(...);

  socket.on(...);
});

io.of("/commands").on("connection", (socket) => {
  socket.join(<device-uuid>);

  io.to(<device-uuid>).emit(...);

  socket.on(...);
});
```

This way, the telemetry events are cleanly split from the commands events.

> Are there any hard/soft limits on the number of Namespaces or Rooms Socket.IO can sensibly handle?

No, there is no special limits.
